### PR TITLE
FPGA Toolchain survey 20241005

### PR DESCRIPTION
**Pull request by Icenowy**
Topic Description
-----------------

- openfpgaloader: update to 0.12.1
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- nextpnr: update to 0.7
    - Enabled Himbaechel for all possible chips.
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- prjoxide: update to newest git
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- prjtrellis: update to newest git
    Needed for newest tagged nextpnr.
    - Skips install RPATH for CMake due to AB4 changes.
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- apicula: update to 0.14
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- python-crc: new, 7.0.0
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
- yosys: update to 0.45
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>

Package(s) Affected
-------------------

- apicula: 0.14
- nextpnr: 1:0.7
- openfpgaloader: 0.12.1
- prjoxide: 0+git20240105
- prjtrellis: 1.4+git20240129
- python-crc: 7.0.0
- yosys: 0.45

Security Update?
----------------

No

Build Order
-----------

```
#buildit yosys python-crc apicula prjtrellis prjoxide nextpnr openfpgaloader
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: app-electronics/nextpnr/autobuild/defines [hunk 1]
```diff
@@ -8,6 +8,9 @@ PKGDES="A portable FPGA place and route tool"
 CMAKE_AFTER="-DARCH=all+alpha \
              -DMISTRAL_ROOT=$SRCDIR/../mistral \
              -DUSE_OPENMP=ON \
-             -DBUILD_GUI=ON"
+             -DBUILD_GUI=ON \
+             -DHIMBAECHEL_GOWIN_DEVICES=all \
+             -DHIMBAECHEL_PRJXRAY_DB=$SRCDIR/../prjxray-db \
+             -DHIMBAECHEL_XILINX_DEVICES=xc7s50;xc7a50t;xc7a100t;xc7a200t;xc7k70t;xc7z010;xc7z020"
```
Comment (MingcongBai) [inline app-electronics/nextpnr/autobuild/defines:15]: Might need a comment explaining this line (or is there an option to enable all supported devices?).

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-electronics/nextpnr/autobuild/defines (hunk 1 above)
Comment (Icenowy) [reply app-electronics/nextpnr/autobuild/defines:15]: Nope, these devices are just what prjxray-db supports now.

Comment (MingcongBai) [reply app-electronics/nextpnr/autobuild/defines:15]: > Nope, these devices are just what prjxray-db supports now.

So maybe add some instructions on how to check for newly supported devices in future updates?

**Comment by Icenowy** [comment]
Smoke tested yosys,nextpnr-himbaechel,apicula,openfpgaloader on Tang Primer 20K by blinky and succeed.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**